src/arrayMethodReduce.js

```diff
@@ -4,8 +4,22 @@
  * Implement method Reduce
  */
 function applyCustomReduce() {
-  [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+  [].__proto__.reduce2 = function (callback, initialValue) {
+    let accum = initialValue;
+    let beginIndex = 0;
+
+    if (arguments.length === 1) {
+      accum = this[0];
+      beginIndex = 1;
+    }
```

Comment:
it's a bit strange that first you assign a possibly undefined value to a variable and only after that you check the arguments.length